### ISSUE TITLE
[DateRangePicker] Month range picker

### ISSUE THREAD

**Issue by joserodolfofreitas** [mui/mui-x#4995]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Sometimes the specific date of the month is not important to the user. In these cases, it'd be interesting to be able to provide a better experience by allowing users to select just the months on the range, like we have with the month view in `DatePicker`.

This issue was originally requested/suggested on a zendesk ticket: https://mui.zendesk.com/agent/tickets/2854

### Examples 🌈

Customer request screenshot. 
![image](https://user-images.githubusercontent.com/550141/170068389-1ef03ed0-ece5-4f13-98cb-1fd69fedebbc.png)

Basic behavior implementation using the simple date picker:
https://codesandbox.io/s/datepicker-views-year-month-l1wyyv?file=/demo.js

**Comment by TheRealCuran** [comment]
In case it helps with prioritising this issue: my Order ID :credit_card: is: 47016

**Comment by spiftire** [comment]
I would like to upwote this issue. my Order ID 💳️ is: 45069

**Comment by destegabry** [comment]
Me too! Order ID 💳 is: 44740

**Comment by razlani** [comment]
Obligatory thumbs up and #metoo please and thank you!

**Comment by istisiki** [comment]
:shipit: please! 🎉

**Comment by razlani** [comment]
> :shipit: please! 🎉

I ended up just building from scratch for month/year range selectors - easier than you think assuming you don't need to leverage some of the fab features of this lib

**Comment by istisiki** [comment]
> I ended up just building from scratch for month/year range selectors - easier than you think assuming you don't need to leverage some of the fab features of this lib

Indeed, we ended up building our own! Would still love it if it was built-in. :)

**Comment by spiftire** [comment]
If you have implemented a solution maybe you wouldn't mind sharing? @razlani @istisiki

**Comment by GoldenDev91** [comment]
I upvote this issue.
This feature is necessary.

**Comment by TheRealCuran** [comment]
@oliviertassinari / @joserodolfofreitas this ticket is still marked as „[waiting for :+1:](https://github.com/mui/mui-x/labels/waiting%20for%20%F0%9F%91%8D)“ while having almost a 150 :+1: and several „votes with wallet“ already. Is this an oversight or are actually more votes needed?

**Comment by joserodolfofreitas** [comment]
@TheRealCuran, 
No more votes are needed. We've planned to work on it just after the date time range components.

**Comment by rsangra** [comment]
+1 Upvote (We are on Pro)

**Comment by razlani** [comment]
> If you have implemented a solution maybe you wouldn't mind sharing? @razlani @istisiki

Would that I could, but for the wrath of my employer.

-Private companies amirite?

**Comment by kinoli** [comment]
Would love to see this implemented.

**Comment by oliviertassinari** [comment]
I have added an example of what it looks like on Google Ads in the issue description.

**Comment by Akbar1909** [comment]
👍

**Comment by mansurmabo** [comment]
Please do it finally my order id 78881. Needs Month, Quarter, Years.

**Comment by michaelfr** [comment]
upvoted !

**Comment by shipsaw** [comment]
Upvoted a well

**Comment by LevapVeeskela** [comment]
Hello, I have DateRangePicker from @mui/x-date-pickers-pro and I resolved custom calendarHeader with two selects for years and months, but when I click on its, then emit event onClose and modal will close... 
I resolved that problem with onOpen and useState, but this is a crutch, and I need use mouseout event for modal as temp a resolve. Can  you add support native events mouseout and etc for container modal/content/paper?
![image](https://github.com/mui/mui-x/assets/32392912/5248ce2e-e107-4905-a7a4-3bc4bd43ca3f)

**Comment by LukasTy** [comment]
@LevapVeeskela Have you tried using `slotProps.popper` to [bind your events](https://stackblitz.com/edit/react-rhz9jp?file=Demo.tsx)? 🤔

**Comment by matheus1519** [comment]
two years later and we don't have a React Month Range Picker from MUI

**Comment by david-ic3** [comment]
Looking for this feature, it's a pain when end users have to go 10 years in the past.

@LukasTy , it is more motivating being focused on new features. But please consolidate what you have. Should we be looking for alternative libraries as we certainly do not want to develop this on our own.

Order ID: #19458

**Comment by LukasTy** [comment]
@david-ic3 I appreciate you voicing your interest and concern! 🙏 
Could you clarify your request?
This issue is about month range picker, but from your comment it seems that the pain point is the lack of year navigation within the Date Range Picker, am I correct? 🤔

**Comment by david-ic3** [comment]
©LukasTy thanks for the quick turnaround. You can check the post of @LevapVeeskela 

When using the date range picker , try selecting a date that is 30 years in the past from the one selected without using the keyboard :-)

<img width="668" alt="image" src="https://github.com/mui/mui-x/assets/80673151/7b1abb19-9915-48f0-bde4-20f6f5f3a29f">

**Comment by RockPaperGardenHose** [comment]
Since you are already working on this feature, would be nice to have MonthRangeCalendar as well :)

**Comment by MatheusCastro99** [comment]
The monthRangePicker / Calendar will be of great help! I will be switching to it as soon as it releases.

**Comment by LukasTy** [comment]
It depends on firstly implementing https://github.com/mui/mui-x/issues/4546.

**Comment by saarbyrner** [comment]
We need this. It is a major blocker for our user base and is table-stakes in most date pickers. We are currently looking at using another library because of the frustration for our users around this.

**Comment by RockPaperGardenHose** [comment]
We need this as well. Blocking major pages from working as intended. 
We are seriously considering unsubscribing from MUI X and look into another library. :(

**Comment by azamatguseinov** [comment]
Same here, major blocker for us, RangePicker is close to useless without this feature. 
I am actually already started looking at the alternatives for this.

**Comment by saarbyrner** [comment]
Not ideal but has anyone tried hacking the v8 version?

**Comment by Vituli** [comment]
I’d like to ask if there are any updates regarding the feature request to support selecting a range of months (instead of specific days) in the Date Range Picker in MUI-X . 
Is there any timeline or plans for development or implementation of this feature?

**Comment by RockPaperGardenHose** [comment]
We would appreciate at least some clarification on if this will be implemented or not. This has been in “next in queue” since the dinosaurs and no explanations on why it has taken so long.

**Comment by RockPaperGardenHose** [comment]
<img width="923" height="963" alt="Image" src="https://github.com/user-attachments/assets/8a0c95d9-c2d2-4ba6-b088-e98bdff5ea53" />


<img width="921" height="961" alt="Image" src="https://github.com/user-attachments/assets/d60f7b48-5fcc-43a9-915a-0a37541d8f22" />

I managed to create my own Month Range Picker (that also supports single month selection) that feels like MUI DateRangePicker but for months... also includes year picker. Hopefully this could be inspiration to some to make their own as well - since this feature is likely not going to be implemented any time soon.

**Comment by naher94** [comment]
This would definitely be useful for our teams as well. If there are any open question regarding interaction patterns happy to help  sort them out.

We have also been investigating a week range selection for reports that run on a weekly schedule for example.

**Comment by ghost** [comment]
I would also like to see this feature 👍

**Comment by apekshit-iksula22** [comment]
i also would like to this feature and thanks in advace MUI team
